pint_xarray/accessors.py

```diff
@@ -400,69 +452,97 @@ class PintDatasetAccessor:
     def __init__(self, ds):
         self.ds = ds
 
-    def quantify(self, units=None, unit_registry=None, registry_kwargs=None):
+    def quantify(
+        self, units=None, unit_registry=None, registry_kwargs=None, **unit_kwargs
+    ):
         """
         Attaches units to each variable in the Dataset.
 
-        Units can be specified as a pint.Unit or as a string, which will
-        be parsed by the given unit registry. If no units are specified then
-        the units will be parsed from the `'units'` entry of the DataArray's
-        `.attrs`. Will raise a ValueError if any of the DataArrays already
-        contain a unit-aware array.
+        Units can be specified as a :py:class:`pint.Unit` or as a
+        string, which will be parsed by the given unit registry. If no
+        units are specified then the units will be parsed from the
+        ``"units"`` entry of the Dataset variable's ``.attrs``. Will
+        raise a ValueError if any of the variables already contain a
+        unit-aware array.
 
         Parameters
         ----------
-        units : mapping from variable names to pint.Unit or str, optional
-            Physical units to use for particular DataArrays in this Dataset. If
-            not provided, will try to read them from
-            `Dataset[var].attrs['units']` using pint's parser.
+        units : mapping of hashable to pint.Unit or str, optional
```

Comment:
This is a user facing function so it would be nice if this said something like `dict (or Mapping) of variable names (Hashable) to pint.Unit or str`

Reply:
I don't think we should do that, that line should be reserved to types and we'd have lots of broken links in the build logs. I'll try to update the description instead.

Reply:
:+1:

Reply:
could you check if this is better?